Warp.WebApp/Data/S3/FileStorage.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Amazon.Runtime.Internal;
+using Amazon.S3;
+using Amazon.S3.Model;
+using System;
+using Warp.WebApp.Models;
+
+namespace Warp.WebApp.Data.S3;
+
+public class FileStorage : IFileStorage
+{
+    private readonly IConfiguration _configuration;
+
+    public FileStorage(IConfiguration configuration)
+    {
+        _configuration = configuration;
+    }
+
+    public async Task SaveFileToStorage(ImageInfo value, CancellationToken cancellationToken)
+    {
+        var s3Client = CreateClient();
+        using (var memoryStream = new MemoryStream(value.Content))
+        {
+            var request = new PutObjectRequest
+            {
+                BucketName = "warp-webapp-dev",
+                Key = value.Id.ToString(),
+                InputStream = memoryStream,
+                ContentType = value.ContentType
+            };
+            var response = await s3Client.PutObjectAsync(request, cancellationToken);
+        }
+    }
+
+    public async Task<ImageInfo> GetFileFromStorage(Guid key, CancellationToken cancellationToken)
+    {
+        var s3Client = CreateClient();
+        var request = new GetObjectRequest
+        {
+            BucketName = "warp-webapp-dev",
+            Key = key.ToString(),
+        };
+
+        var response = await s3Client.GetObjectAsync(request, cancellationToken);
+        if (response != null)
```

Comment:
If one of condition branches is complex, better keep the other one complex too. Otherwise, you may invert the condition:
```csharp
if (response == null)
    return default;

using var memoryStream = new MemoryStream();
response.ResponseStream.CopyTo(memoryStream);
return new ImageInfo
    {
        Id = key,
        Content = memoryStream.ToArray(),
        ContentType = response.Headers.ContentType
    };
```